Data metrics - Packet Loss and Packets dropped metrics

Ketant0 opened the pull request:
Added packet loss related metrics. Showing the percent dropped and also the count of the dropped packets in the status bar. It looks like this in the end. 

<img width="1344" height="720" alt="Screenshot 2025-09-11 at 4 20 44 PM" src="https://github.com/user-attachments/assets/28504593-2eab-4512-8547-4b7722608406" />

joshua-t0 commented:
Right now the behavior is the the packet loss is calculated as a percentage of all time.

So if i do something that causes packet loss the dropped packet spikes, and then stops, and the packet loss percentage spikes, and then slowly over time decreases in percentage.

This is formally logical, but I think confusing if we are imagining how people will uses this.

I expect people want to keep track of dropped statistics and also of packet loss over some reasonable period of time.
For example, if i change some network configuration after this has been running for a long time, even if my dropped packet counter climbs because i am now losing 50% of my packets, the packet loss percentage will only slowly climb upward asymptotically.

We should also provide a few more user prompts here.

1. Is it possible for the packet loss text to turn red if it's over 1%?
2. Is it possible to have a thing that says "CONSULT HELP FOR NETWORKING SUGGESTIONS" if packet loss is high?
3. We should have the "Help" dialogue include the steps to improve network issues that we know of on each of the OS systems.

Ketant0 commented:
Implemented all the changes suggested above. 

Added the steps I knew that can help reduce packet loss in the "Help" dialog box. We can add more info there. I have left the Windows related part blank for now as I am working on it. 

It looks like this now - 
<img width="1321" height="684" alt="Screenshot 2025-09-16 at 11 27 10 AM" src="https://github.com/user-attachments/assets/e02530f8-b807-4151-9f40-634968fda0f7" />

Ketant0 commented:
Fix implemented, it picks up default color as per the Windows color.

joshua-t0 commented:
Hi Ketan,

This does look like it calculates the packet loss on a rolling basis now, and I see the red and the warning when packet loss becomes significant, but once the packet loss has returned to normal, the red text and the message about high packet loss still persists.

<img width="1385" height="702" alt="image" src="https://github.com/user-attachments/assets/bd2e5b58-3bef-4ad0-bd85-6fdca7412aa9" />

Ketant0 commented:
Thanks for catching this! I assumed it will pick up the default color again but that is not the case with QtGui. It sets "red" as the default color after percent > 1 condition is executed.

I fixed it by initializing the default color before the loop even starts and just call that value.

Also added command to remove the Info message.

Tested both on Mac and Linux, works as expected now. Let me know if there is still any issue.

Ketant0 commented on rfmux/tools/periscope/app.py:
````diff
@@ -735,6 +753,20 @@ def _show_help(self):
+            "```\n"
+            "     sudo net.core.rmem_max = 67108864\n"
+            "```\n"
+            "- **Windows:** \n\n"
````
Yeah I did come up with solution. But I am not sure how effective they are since I did not have the windows implementation in this branch. Will use it once Periscope on Windows branch is merged into main.
